src/aggregator/aggregator/aggregator.go

```diff
@@ -219,20 +220,39 @@ func (agg *aggregator) placementTick() {
 }
 
 func (agg *aggregator) AddUntimed(
-	metric unaggregated.MetricUnion,
+	union unaggregated.MetricUnion,
 	metadatas metadata.StagedMetadatas,
 ) error {
+	if agg.opts.TimedForResendEnabled() {
+		for _, m := range metadatas {
+			for _, p := range m.Pipelines {
```

Comment:
```suggestion
	if agg.opts.TimedForResendEnabled() {
		for i := range metadatas {
			for j := range m[i].Pipelines {
			...
```

Reply:
these structs can be quite big, I suggest iterating over indexes and not copying over values in each loop

Reply:
good call!